libs/core/src/lib/splitter/splitter-pane-container/splitter-pane-container.component.ts

```diff
@@ -0,0 +1,439 @@
+                })
+        )
+
+        if (this._document) {
+            this._subscription$.add(
+                fromEvent(this._document.defaultView, 'resize')
+                    .subscribe(() => this._resizePanesToFitInContainer())
+            )
+        }
+    }
+
+    /** @hidden */
+    private _setDefaultPane(): void {
+        const setDefaultPane = (): void => {
+            if (this._splitter.defaultPaneId) {
+                this._defaultPane = this._panes.find(pane => pane.id === this._splitter.defaultPaneId);
+            }
+
+            if (!this._defaultPane) {
+                this._defaultPane = this._directPanes[0];
+            }
+
+            /** Detaching basically works without the code but it's the edge-case */
+            if (this._defaultPane?._content.isAttached) {
+                this._defaultPane._content.detach();
+            }
+        }
+
+        setDefaultPane();
+
+        this._subscription$.add(
+            this._splitter._defaultPaneId$
+                .subscribe(() => {
+                    this._cdr.detectChanges();
+
+                    setDefaultPane();
+
+                    this._resizePanesToFitInContainer(true);
+                    this._updatePages();
+                })
+        );
+    }
+
+    /** @hidden */
+    private _resizePanesToFitInContainer(setInitialSize = false): void {
+        if (setInitialSize) {
+            this._cdr.detectChanges();
+        }
+
+        const containerSizePx = this._isHorizontal
+            ? this._elementRef.nativeElement.offsetHeight
+            : this._elementRef.nativeElement.offsetWidth;
+        let availableSpacePx = containerSizePx - RESIZER_SIZE_PX * (this._directPanes.length - 1);
+
+        this._directPanes.forEach((pane, index) => {
+            if (!availableSpacePx) {
+                pane._actualSize = '0px';
+                return;
+            }
+
+            if (setInitialSize) {
+                this._wantedPanesSizes[index] = this._getPaneElementSizePx(pane.id);
+            }
+
+            const paneSize = this._wantedPanesSizes[index];
+
+            if (availableSpacePx - this._wantedPanesSizes[index] < 0) {
+                pane._actualSize = availableSpacePx + 'px';
+                availableSpacePx = 0;
+                return;
+            }
+
+            pane._actualSize = pane.size === PANE_AUTO_SIZE ? PANE_AUTO_SIZE : (paneSize + 'px');
+            availableSpacePx -= paneSize;
+        });
+
+        this._cdr.detectChanges();
+    }
+
+
+    /** @hidden */
+    private _getPaneElementSizePx(paneId: string): number {
+        const paneElement = this._getPaneElement(paneId);
+
+        if (!paneElement) {
+            return null;
+        }
+
+        return this._isHorizontal ? paneElement.offsetHeight : paneElement.offsetWidth;
+    }
+
+    /** @hidden */
+    private _getPaneElement(paneId: string): HTMLElement {
+        return this._elementRef.nativeElement.querySelector(`#${paneId}`);
+    }
+}
```

Comment:
As for me it's so big component, It would be great to delegate some logic to some directive or service if it's possible. But up to you, it's just recommendation.

Reply:
Agree but the code is highly tied up with `@ContentChildren` due to the component's functionality and in the case of moving methods to the service we also need to manage the various panes state there. 

Splitting into the base & component (extends `base`) seems not the case here as then base will act as the `container` and component as the `root container` but we're exposing only one component to the customers and it should be named just a `container` to not confuse anyone.